// mat-sidenav-menu/esm5/lib/expansion-group.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var SideNavExpansionGroupComponent = /** @class */ (function () {
    function SideNavExpansionGroupComponent() {
    }
    SideNavExpansionGroupComponent.decorators = [
        { type: Component, args: [{
                    selector: 'side-nav-x-group',
                    template: "<mat-expansion-panel [hideToggle]=\"hideToggle\">\n        <mat-expansion-panel-header *ngIf=\"title\">{{ title }}</mat-expansion-panel-header>\n        <ng-content></ng-content>\n    </mat-expansion-panel>",
                    styles: ["\n    div.mat-expansion-panel-body {\n      padding-right: 0px;\n      padding-left: 0px;\n    }\n  "]
                }] }
    ];
    SideNavExpansionGroupComponent.propDecorators = {
        title: [{ type: Input }],
        hideToggle: [{ type: Input }]
    };
    return SideNavExpansionGroupComponent;
}());
export { SideNavExpansionGroupComponent };
if (false) {
    /** @type {?} */
    SideNavExpansionGroupComponent.prototype.title;
    /** @type {?} */
    SideNavExpansionGroupComponent.prototype.hideToggle;
}
