// mat-sidenav-menu/esm2015/lib/item.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
export class SideNavItemComponent {
    /**
     * @param {?} router
     */
    constructor(router) {
        this.router = router;
    }
    /**
     * @return {?}
     */
    navigate() {
        if (!this.disabled) {
            this.router.navigate([this.route]);
        }
    }
    /**
     * @param {?} path
     * @return {?}
     */
    activatePath(path) {
        if (path === this.route) {
            this.active = true;
        }
        else {
            this.active = false;
        }
    }
}
SideNavItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'side-nav-item',
                template: `<li>
                  <a class="side-nav-item" matRipple
                    [ngClass]="{'side-nav-item-active': active,'side-nav-item-disabled': disabled}"
                    (click)="navigate()">
                        <mat-icon *ngIf="icon" style="margin-right:10px;vertical-align:middle">{{ icon }}</mat-icon>
                        <span style="vertical-align:middle">{{ title }}</span>
                  </a>
               </li>`,
                styles: [`.side-nav-item {
          list-style-type: none;
          box-sizing: border-box;
          display: block;
          font-size: 14px;
          font-weight: 400;
          line-height: 47px;
          text-decoration: none;
          transition: all .3s;
          padding: 0 16px;
          position: relative;
          cursor: pointer;
        }`]
            }] }
];
/** @nocollapse */
SideNavItemComponent.ctorParameters = () => [
    { type: Router }
];
SideNavItemComponent.propDecorators = {
    title: [{ type: Input }],
    icon: [{ type: Input }],
    route: [{ type: Input }],
    disabled: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    SideNavItemComponent.prototype.title;
    /** @type {?} */
    SideNavItemComponent.prototype.icon;
    /** @type {?} */
    SideNavItemComponent.prototype.route;
    /** @type {?} */
    SideNavItemComponent.prototype.disabled;
    /** @type {?} */
    SideNavItemComponent.prototype.active;
    /** @type {?} */
    SideNavItemComponent.prototype.router;
}
