// mat-sidenav-menu/esm2015/lib/mat-sidenav-menu.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import { Component, ContentChildren, Input, QueryList } from '@angular/core';
import { Router, ActivatedRoute, NavigationEnd } from '@angular/router';
import { MatSidenav } from '@angular/material';
import { SideNavItemComponent } from './item';
export class MatSidenavMenuComponent {
    /**
     * @param {?} router
     * @param {?} aroute
     */
    constructor(router, aroute) {
        this.router = router;
        this.aroute = aroute;
        // console.log('router:', this.router);
        this.router.events.subscribe(e => {
            // console.log('url:', this.router.url);
            // console.log('router state:', this.router.routerState);
            // if (this.router.routerState.root.children.length > 0 ) {
            if (e instanceof NavigationEnd) {
                // console.log('component:', '(' + Date.now + ')' + this.router.routerState.root.children[0].component);
                // console.log('path: ' + this.getPath());
                this.items.forEach(i => {
                    i.activatePath(this.getPath());
                });
                this.sidenav.close();
            }
            // console.log('path:', this.erouterState);
        });
        // this.aroute.url.subscribe( (u) => {
        //   console.log('active route:', u);
        // });
    }
    /**
     * @return {?}
     */
    getPath() {
        /** @type {?} */
        const path = this.router.routerState.snapshot.url;
        return path;
    }
}
MatSidenavMenuComponent.decorators = [
    { type: Component, args: [{
                selector: 'side-nav-menu',
                template: `
      <nav style="height:100%">
        <mat-toolbar color="primary">
          <button mat-icon-button (click)="sidenav.close()"><mat-icon>keyboard_arrow_left</mat-icon></button>&nbsp;
          <span>{{ title }}</span>
        </mat-toolbar>

        <ul class="side-nav-menu">
          <ng-content></ng-content>
        </ul>
      </nav>`,
                styles: [`.side-nav-menu {
          list-style-type: none;
          padding: 0
      }`]
            }] }
];
/** @nocollapse */
MatSidenavMenuComponent.ctorParameters = () => [
    { type: Router },
    { type: ActivatedRoute }
];
MatSidenavMenuComponent.propDecorators = {
    sidenav: [{ type: Input }],
    title: [{ type: Input }],
    items: [{ type: ContentChildren, args: [SideNavItemComponent, { descendants: true },] }]
};
if (false) {
    /** @type {?} */
    MatSidenavMenuComponent.prototype.sidenav;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.title;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.items;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.active;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.router;
    /** @type {?} */
    MatSidenavMenuComponent.prototype.aroute;
}
